filters/builtin/stripquery.go

```diff
@@ -62,6 +63,18 @@ func sanitize(input string) string {
 	return b.String()
 }
 
+// temporary public function to benchmark it
+func NewSanitize(input string) string {
+	toAscii := strconv.QuoteToASCII(input)
+	var s strings.Builder
+	for _, i := range toAscii {
+		if validHeaderFieldByte(byte(i)) {
+			s.WriteRune(i)
+		}
+	}
+	return s.String()
```

Comment:
To get ultimate performance we can use byte version https://pkg.go.dev/strconv#AppendQuoteToASCII here and then replace invalid bytes with `-` instead of allocating another slice.